Sources/TensorFlow/Operators/NN.swift

```diff
@@ -184,8 +184,8 @@ func _vjpConv2DBackpropInput<Scalar: TensorFlowFloatingPoint>(
 ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> (Tensor<Scalar>, Tensor<Scalar>)) {
     let value = conv2DBackpropInput(x, shape: shape, filter: filter,
                                     strides: strides, padding: padding, dilations: dilations)
-    return (value, { v in
-        (conv2DBackpropFilter(x, input: v, filterSizes: shape, strides: strides,
+    return (value, { [filterShape = filter.shapeTensor] v in
```

Comment:
Actually this pullback is already capturing `filter` (on line 190), so I think your original implementation is better! I reverted it for you.

Reply:
Thanks!